services/app-api/src/resolvers/contractResolvers/updateContractResolver.ts

```diff
@@ -0,0 +1,110 @@
+import { ForbiddenError, UserInputError } from 'apollo-server-lambda'
+import {
+    isCMSUser,
+    convertContractWithRatesToUnlockedHPP,
+} from '../../domain-models'
+import type { MutationResolvers } from '../../gen/gqlServer'
+import { logError } from '../../logger'
+import type { Store } from '../../postgres'
+import {
+    setResolverDetailsOnActiveSpan,
+    setErrorAttributesOnActiveSpan,
+} from '../attributeHelper'
+import type { LDService } from '../../launchDarkly/launchDarkly'
+import { GraphQLError } from 'graphql'
+
+export function updateContractResolver(
+    store: Store,
+    launchDarkly: LDService
+): MutationResolvers['updateContract'] {
+    return async (_parent, { input }, context) => {
+        const { user, span } = context
+        setResolverDetailsOnActiveSpan('updateContract', user, span)
+        const ratesDatabaseRefactor = await launchDarkly.getFeatureFlag(
+            context,
+            'rates-db-refactor'
+        )
+
+        if (ratesDatabaseRefactor) {
+            // This resolver is only callable by CMS users
+            if (!isCMSUser(user)) {
+                logError(
+                    'updateContract',
+                    'user not authorized to update contract'
+                )
+                setErrorAttributesOnActiveSpan(
+                    'user not authorized to update contract',
+                    span
+                )
+                throw new ForbiddenError(
+                    'user not authorized to update contract'
+                )
+            }
+
+            const contract = await store.findContractWithHistory(input.id)
+            if (contract instanceof Error) {
+                throw contract
+            }
+
+            const hasSubmitInfo =
+                contract.revisions.filter((rev) => rev.submitInfo).length > 0
+
+            if (!hasSubmitInfo) {
+                const errMessage = `Can not update a contract has not been submitted. Fails for contract with ID: ${contract.id}`
+                logError('updateContract', errMessage)
+                setErrorAttributesOnActiveSpan(errMessage, span)
+                throw new UserInputError(errMessage, {
```

Comment:
In my mind this resolver encountering a contract in a non submitted status would be a 409 HTTP conflict error. I'm not sure if `UserInputError` is the correct error to throw in this case?

Reply:
Right now we're using UserInputError for basically all 4XX errors. If it's an error where the user made a mistake that's a UserInputError. If it's an unexpected server error to an otherwise valid request, that's an Error()